### PR TITLE
[doc][build/01] BuildCache api class

### PR DESCRIPTION
**Pull request by can-anyscale**
Move the logic of getting and zipping the doc build artifacts into a class, and add test cases. Subsequently, I'll add the logic to modify the pickle file to remove things such as site-packages dependency, before uploading.

Test:
- CI

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
(for @khluu to review, @khluu let me know if you need help)

File: ci/ray_ci/doc/build_cache.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        return {file for file in untracked_files if file}
+
+    def _zip_cache(self, cache_files: Set[str]) -> str:
+        """
+        Create a tarball of the cache files
+        """
+        with tempfile.NamedTemporaryFile(mode="w+t") as temp_file:
+            temp_file.write("\n".join(cache_files))
+            doc_tarball = f'{os.environ["BUILDKITE_COMMIT"]}.tgz'
```
Comment (khluu) [inline ci/ray_ci/doc/build_cache.py:64]: can we have this as full absolute path?

Comment (khluu) [reply ci/ray_ci/doc/build_cache.py:64]: maybe `os.path.join(<path_to_ray_dir>, f'{os.environ["BUILDKITE_COMMIT"]}.tgz')` or something

**Review by khluu — APPROVED** [review]

File: ci/ray_ci/doc/build_cache.py (hunk 1 above)
Comment (khluu) [reply ci/ray_ci/doc/build_cache.py:64]: or using `_cache_dir`

**Review by khluu — APPROVED** [review]

**Pull request merged**